tests/compliance.js

```diff
@@ -245,7 +245,7 @@ function test_selection_compliance (SelectionClass, setup=function () {}, extra_
             // Not all Selections might use the Xapian engine, but I can't think
             // of a better place to put this
             let engine = MockEngine.mock_default();
-            engine.query_finish.and.returnValue({ models: [] });
+            engine.query_promise.and.returnValue(Promise.resolve({ models: [] }));
```

Comment:
Nothing wrong with this, but out of curiosity, I suppose we could just leave this as-is and write an implementation of `MockEngine.query_promise()` to call `query_finish()`?